Libs/MRML/Logic/vtkMRMLSliceLogic.cxx

```diff
@@ -1022,6 +1033,706 @@ void vtkMRMLSliceLogic::UpdateReconstructionSlab(vtkMRMLSliceLogic* sliceLogic,
+                  << resampleScalarVectorDWIVolumeString << " failed to run");
+    return false;
+  }
+
+  outputStraightenedVolume->CreateDefaultDisplayNodes();
+  vtkMRMLDisplayNode* volumeDisplayNode = volumeNode->GetDisplayNode();
+  if (volumeDisplayNode)
+  {
+    outputStraightenedVolume->GetDisplayNode()->CopyContent(volumeDisplayNode);
+  }
+  return true;
+}
+
+//----------------------------------------------------------------------------
+bool vtkMRMLSliceLogic::CurvedPlanarReformationProjectVolume(vtkMRMLScalarVolumeNode* outputProjectedVolume,
+                                                             vtkMRMLScalarVolumeNode* inputStraightenedVolume,
+                                                             int projectionAxisIndex)
+{
+  // Create panoramic volume by mean intensity projection along an axis of the straightened volume
+
+  if (outputProjectedVolume == nullptr)
+  {
+    vtkErrorMacro("vtkMRMLSliceLogic::CurvedPlanarReformationProjectVolume failed: "
+                  "outputProjectedVolume not supplied");
+    return false;
+  }
+  if (inputStraightenedVolume == nullptr)
+  {
+    vtkErrorMacro("vtkMRMLSliceLogic::CurvedPlanarReformationProjectVolume failed: "
+                  "inputStraightenedVolume not supplied");
+    return false;
+  }
+  if ((projectionAxisIndex < 0) || (projectionAxisIndex >= 3))
+  {
+    vtkErrorMacro("vtkMRMLSliceLogic::CurvedPlanarReformationProjectVolume failed: "
+                  "projectionAxisIndex is out of range");
+    return false;
+  }
+
+  // Create a new vtkImageData for the projected volume
+  vtkSmartPointer<vtkImageData> projectedImageData = vtkSmartPointer<vtkImageData>::New();
+  outputProjectedVolume->SetAndObserveImageData(projectedImageData);
+
+  // Get the image data from the input straightened volume
+  vtkImageData* straightenedImageData = inputStraightenedVolume->GetImageData();
+  if (!straightenedImageData)
+  {
+    vtkErrorMacro(
+      "vtkMRMLSliceLogic::CurvedPlanarReformationProjectVolume failed: input straightened volume must have image data");
+    return false;
+  }
+
+  // Get the dimensions of the straightened volume
+  int outputImageDimensions[3] = { 0, 0, 0 };
+  straightenedImageData->GetDimensions(outputImageDimensions);
+  outputImageDimensions[projectionAxisIndex] = 1; // Set the projection axis to size 1
+  projectedImageData->SetDimensions(outputImageDimensions);
+
+  // Allocate scalars for the projected image
+  projectedImageData->AllocateScalars(straightenedImageData->GetScalarType(),
+                                      straightenedImageData->GetNumberOfScalarComponents());
+
+  // Get arrays of the input and output volumes
+  vtkSmartPointer<vtkDataArray> outputProjectedVolumeArray = projectedImageData->GetPointData()->GetScalars();
+  vtkSmartPointer<vtkDataArray> inputStraightenedVolumeArray = straightenedImageData->GetPointData()->GetScalars();
```

Comment:
```suggestion
  vtkDataArray* outputProjectedVolumeArray = projectedImageData->GetPointData()->GetScalars();
  vtkDataArray* inputStraightenedVolumeArray = straightenedImageData->GetPointData()->GetScalars();
```

Any specific reason for using `vtkSmartPointer` only for those ?

Reply:
I will make the change